File: tests/algorithms/x/Java/maths/gaussian.java
public class Main {
    static double PI;

    static double sqrtApprox(double x) {
        double guess = x / 2.0;
        long i_1 = 0;
        while (i_1 < 20) {
            guess = (guess + x / guess) / 2.0;
            i_1 = i_1 + 1;
        }
        return guess;
    }

    static double expApprox(double x) {
        boolean is_neg = false;
        double y_1 = x;
        if (x < 0.0) {
            is_neg = true;
            y_1 = -x;
        }
        double term_1 = 1.0;
        double sum_1 = 1.0;
        long n_1 = 1;
        while (n_1 < 30) {
            term_1 = term_1 * y_1 / (((Number)(n_1)).doubleValue());
            sum_1 = sum_1 + term_1;
            n_1 = n_1 + 1;
        }
        if (is_neg) {
            return 1.0 / sum_1;
        }
        return sum_1;
    }

    static double gaussian(double x, double mu, double sigma) {
        double coeff = 1.0 / sqrtApprox(2.0 * PI * sigma * sigma);
        double exponent_1 = -((x - mu) * (x - mu)) / (2.0 * sigma * sigma);
        return coeff * expApprox(exponent_1);
    }

    static void main() {
        double result = gaussian(1.0, 0.0, 1.0);
        System.out.println(result);
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            PI = 3.141592653589793;
            main();
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }
}

File: tests/algorithms/x/Kotlin/maths/gaussian.kt
var PI: Double = 3.141592653589793
fun sqrtApprox(x: Double): Double {
    var guess: Double = x / 2.0
    var i: Int = (0).toInt()
    while (i < 20) {
        guess = (guess + (x / guess)) / 2.0
        i = i + 1
    }
    return guess
}

fun expApprox(x: Double): Double {
    var is_neg: Boolean = false
    var y: Double = x
    if (x < 0.0) {
        is_neg = true
        y = 0.0 - x
    }
    var term: Double = 1.0
    var sum: Double = 1.0
    var n: Int = (1).toInt()
    while (n < 30) {
        term = (term * y) / ((n.toDouble()))
        sum = sum + term
        n = n + 1
    }
    if ((is_neg as Boolean)) {
        return 1.0 / sum
    }
    return sum
}

fun gaussian(x: Double, mu: Double, sigma: Double): Double {
    var coeff: Double = 1.0 / sqrtApprox(((2.0 * PI) * sigma) * sigma)
    var exponent: Double = (0.0 - ((x - mu) * (x - mu))) / ((2.0 * sigma) * sigma)
    return coeff * expApprox(exponent)
}

fun user_main(): Unit {
    var result: Double = gaussian(1.0, 0.0, 1.0)
    println(result)
}

fun main() {
    user_main()
}
